src/js/components/Stack/Stack.js

```diff
@@ -1,69 +1,82 @@
-import React, { Children, Component } from 'react';
+import React, { Children } from 'react';
 
 import { StyledStack, StyledStackLayer } from './StyledStack';
 
-class Stack extends Component {
-  toChildIndex = child => {
-    const { children } = this.props;
-    let index = child;
-    if (index === 'first' || !index) {
-      index = 0;
-    } else if (index === 'last') {
-      index = React.Children.count(children) - 1;
+const buildStyledChildren = ({
+  fill,
+  guidingIndex,
+  interactiveIndex,
+  interactiveChild,
+  anchor,
+}) => {
+  let childIndex = 0;
+
+  return child => {
+    if (child) {
+      const interactive =
+        interactiveChild === undefined || interactiveIndex === childIndex;
+
+      const isGuidingIndex = childIndex === guidingIndex;
+      childIndex += 1;
+
+      const props = isGuidingIndex
+        ? {
+            guiding: true,
+            fillContainer: fill,
+          }
+        : {
+            anchor,
+          };
+
+      return (
+        <StyledStackLayer interactive={interactive} {...props}>
+          {child}
+        </StyledStackLayer>
+      );
     }
-    return index;
+
+    return child;
   };
+};
 
-  render() {
-    const {
-      anchor,
-      children,
-      fill,
-      guidingChild,
-      interactiveChild,
-      ...rest
-    } = this.props;
+const toChildIndex = (child, children) => {
```

Comment:
What are your thoughts on moving `toChildIndex` inside of the functional component of Stack, and remove the arguments of `child` and `children`?

Reply:
I think is not a good ideia, because Stack is calling toChildIndex twice with differents child argument. I can move toChildIndex to inside of the Stack component and remove the second argument, but in my opinion is not the best solution.

Reply:
Thanks for your inputs, I overlooked the fact that `child` should remain an argument, but I still think we can move the function inside of Stack, `children` could be removed and the only argument would be `child`. why would you think this is not a good idea?

Reply:
In my opinion the better solution always is the component just render the element, all the others logics should be outside the component, but I can change, no problems.

Reply:
Done :blush: